### PR TITLE
Renamed Bib to WithTileFootprintOverlay

### PR DESCRIPTION
**Pull request by reaperrr**
Split from #9390.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs [hunk 1]
```diff
@@ -15,17 +15,17 @@
...
 		[PaletteReference] public readonly string Palette = "terrain";
 		public readonly bool HasMinibib = false;
 
-		public object Create(ActorInitializer init) { return new Bib(init.Self, this); }
+		public object Create(ActorInitializer init) { return new WithTileFootprintOverlay(init.Self, this); }
 
 		public IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer init, RenderSpritesInfo rs, string image, int facings, PaletteReference p)
 		{
-			if (init.Contains<HideBibPreviewInit>() && init.Get<HideBibPreviewInit, bool>())
+			if (init.Contains<HideWithTileFootprintOverlayPreviewInit>() && init.Get<HideWithTileFootprintOverlayPreviewInit, bool>())
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs]: > Can we add a flag or another way to turn this off? This trait has other uses (such as the dune 2 structures and bridges), and they won't want to be affected by this.

File: OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs [hunk 2]
```diff
@@ -15,17 +15,17 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
-	public class BibInfo : ITraitInfo, Requires<BuildingInfo>, IRenderActorPreviewSpritesInfo, Requires<RenderSpritesInfo>
+	public class WithTileFootprintOverlayInfo : ITraitInfo, Requires<BuildingInfo>, IRenderActorPreviewSpritesInfo, Requires<RenderSpritesInfo>
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs]: > We could make this much more general purpose by defining a Size and Offset here, and then we can remove the Minibib stuff (which is just a normal bib but with a different size and offset).

Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs]: :+1:

File: OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs (hunk 1 above)
Comment (atlimit8) [inline OpenRA.Mods.Common/Traits/Render/WithTileFootprintOverlay.cs]: Sounds reasonable.

**Comment by atlimit8** [comment]
Needs to be rebased.

**Comment by abcdefg30** [comment]
No activity since a while, please reopen this when you come back to it.

**Pull request closed without merging**